src/Mvc/Mvc.RazorPages/src/PageBase.cs

```diff
@@ -63,6 +64,7 @@ public abstract class PageBase : RazorPageBase
     /// <summary>
     /// Gets or sets the <see cref="IModelMetadataProvider"/>.
     /// </summary>
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
     public IModelMetadataProvider MetadataProvider
```

Comment:
One thing to note is that, even though some of these properties are not "directly" useful. They are indirectly useful in many cases. For example, when you try to evaluate an expression in the watch window or trying to inspect some related piece of state. Something to consider, as this affects "discoverability" during debugging. I assume they can still be accessed from the debugger, isn't it?

Reply:
Lets discuss in https://github.com/dotnet/aspnetcore/pull/49353#discussion_r1261049746